posix/stdlib.c

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/corelibc/errno.h>
+#include <openenclave/corelibc/limits.h>
+#include <openenclave/corelibc/stdio.h>
+#include <openenclave/corelibc/stdlib.h>
+#include <openenclave/corelibc/string.h>
+#include <openenclave/corelibc/unistd.h>
+#include <openenclave/internal/posix/raise.h>
+#include <openenclave/internal/trace.h>
+
+char* oe_realpath(const char* path, char resolved_path[OE_PATH_MAX])
+{
+    char* ret = NULL;
+    char buf[OE_PATH_MAX];
+    const char* in[OE_PATH_MAX];
+    size_t nin = 0;
+    const char* out[OE_PATH_MAX];
+    size_t nout = 0;
+    char resolved[OE_PATH_MAX];
```

Comment:
We are allocating a lot of 4KB buffers on the stack here (5x if you include `cwd`) We should switch to heap allocs at this point.

Reply:
Fair enough. Created local **variable_t* struct an put on the heap.